### PR TITLE
nnbd migration

### PR DESCRIPTION
**Pull request by pq**
First crack at a migration.

Feedback welcome!

/cc @devoncarew @lrhn @natebosch

### CODE REVIEW

File: lib/cli_logging.dart [hunk 1]
```diff
@@ -161,20 +158,16 @@ class StandardLogger implements Logger {
   }
 
   void _cancelProgress() {
-    if (_currentProgress != null) {
-      var progress = _currentProgress;
+    var progress = _currentProgress;
+    if (progress != null) {
```
Comment (srawlins) [inline lib/cli_logging.dart:162]: Ha, clever flip.

**Review by srawlins — APPROVED** [review]
Wow I love it.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,14 +1,15 @@
 name: cli_util
-version: 0.2.1-dev
+version: 0.3.0-nullsafety
```
Comment (devoncarew) [inline pubspec.yaml:2]: Is the convention here `0.3.0-nullsafety` or `0.3.0-nullsafety.1`?

File: pubspec.yaml [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 dependencies:
-  path: '>=1.0.0 <2.0.0'
+  meta: ^1.3.0-nullsafety
+  path: ^1.8.0-nullsafety
 
 dev_dependencies:
-  pedantic: ^1.0.0
-  test: ^1.0.0
+  pedantic: ^1.10.0-nullsafety
+  test: ^1.16.0-nullsafety
```
Comment (devoncarew) [inline pubspec.yaml:15]: I didn't know there was a null safe version of `test`.

**Review by devoncarew — APPROVED** [review]
👍 

I assume we publish these after 2.12.0- is out?

File: pubspec.yaml (hunk 2 above)
Comment (pq) [reply pubspec.yaml:2]: No idea to be honest!

Comment (pq) [reply pubspec.yaml:2]: The first place I looked has *both* conventions:

https://github.com/dart-lang/path/blob/7dedbbe99e76f7cd154e4ac26db3e9a183c3525b/pubspec.yaml#L14-L15

🤷 

@natebosch: any opinion?

Comment (pq) [reply pubspec.yaml]: Offline conversation has lead to `0.3.0-nullsafety` as the preferred convention.

**Pull request merged**